readthedocs/telemetry/collectors.py

```diff
@@ -62,8 +69,8 @@ def collect(self):
         )
         data["packages"] = {
             "pip": {
-                "user": self._get_pip_packages(include_all=False),
-                "all": self._get_pip_packages(include_all=True),
+                "user": self._get_user_pip_packages(),
+                "all": self._get_all_pip_packages(),
```

Comment:
This feels really brittle. I'd love to have a bit more protection here against exceptions, given that we're running a ton of code here, interacting with third-party modules and the filesystem, which feels highly likely to break or time out..

Reply:
The whole code is ran inside a try/except block from outside because of this concern specifically: https://github.com/readthedocs/readthedocs.org/blob/100bf6c08e16c2917272b8b1e4ff816c38616dda/readthedocs/projects/tasks/builds.py#L694-L706

So, I think we are already protected about this problem.

Reply:
Yea, it just means that if one thing breaks, all the data is lost, but I guess that's probably fine?

Reply:
Oh, I see. Yes, this is how it has been working so far. We could wrap each command on a try except block to always save as much data as we can. However, we haven't seen this problem so far. I will add a note in the code.